src/lib/modules/ledger/services/LedgerService.ts

```diff
@@ -182,6 +185,29 @@ export class LedgerService {
     }
   }
 
+  private async submitWriteRequest(request: LedgerRequest, signDid: string): Promise<LedgerWriteReplyResponse> {
+    request = await this.appendTaa(request);
+    request = await this.wallet.signRequest(signDid, request);
```

Comment:
Doesn't this mutate the `request` param? What about naming that like `requestWithTaa` and `signedRequestWithTaa`? Or do you think it's too much and chatty? :)

Reply:
It reassigns. So yeah it overwrites the request param, but it doesn't change the value outside of the method